File: backend/src/main/java/com/backend/bikerental/entity/VehicleBrand.java
package com.backend.bikerental.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "vehicle_brands")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class VehicleBrand {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id;
    @Column(unique = true, nullable = false)
    String name;
    String country;
    @CreationTimestamp
    @Column(nullable = false)
    LocalDateTime createdAt;
}

File: frontend/packages/types/src/vehicleBrand.ts
export interface VehicleBrand {
  id: number;
  name: string;
  country: string;
  createdAt: string;
}

export interface VehicleBrandCreationRequest {
  name: string;
  country: string;
}

export interface VehicleBrandUpdateRequest {
  name: string;
  country: string;
}
